### ISSUE TITLE
[Bug]: Crash on NeoForge 21.0.30-beta

### ISSUE THREAD

**Issue by thomasglasser** [SSKirillSS/Curios#1]
### Minecraft Version

1.20.6

### What happened?

On block break:
```
java.lang.NoSuchMethodError: 'int net.minecraft.world.level.block.state.BlockState.getExpDrop(net.minecraft.world.level.LevelReader, net.minecraft.util.RandomSource, net.minecraft.core.BlockPos)'
at TRANSFORMER/curios@9.0.2+1.21/top.theillusivec4.curios.common.event.CuriosEventHandler.onBreakBlock(CuriosEventHandler.java:478)
```
This seems to be because the method signature has been changed.

### How do you trigger this bug?

1. Break a block in NeoForge 21.0.30-beta

### Loader

NeoForge

### Loader Version

NeoForge 21.0.30-beta

### Mod Version

9.0.2+1.21

### Relevant Log Outputs

```
java.lang.NoSuchMethodError: 'int net.minecraft.world.level.block.state.BlockState.getExpDrop(net.minecraft.world.level.LevelReader, net.minecraft.util.RandomSource, net.minecraft.core.BlockPos)'
at TRANSFORMER/curios@9.0.2+1.21/top.theillusivec4.curios.common.event.CuriosEventHandler.onBreakBlock(CuriosEventHandler.java:478)
```

**Comment by thomasglasser** [comment]
Also 1.21 isnt an option on the form

**Comment by ashton727** [comment]
same issue

> Also 1.21 isnt an option on the form

**Comment by UnknownSerhan** [comment]
Same problem, easy to reproduce

**Comment by SSKirillSS** [comment]
Sorry for the late response! Should be fixed since 9.0.5 release.
